src/requests/cache/scalar/keys_exist.rs

```diff
@@ -0,0 +1,106 @@
+use crate::utils::parse_string;
+use crate::{CacheClient, IntoBytes, MomentoResult};
+
+/// Request to check if the provided keys exist in the cache.
+/// Returns a list of booleans indicating whether each given key was found in the cache.
+///
+/// # Arguments
+/// * `cache_name` - name of cache
+/// * `keys` - list of keys to look up
+///
+/// # Examples
+/// Assumes that a CacheClient named `cache_client` has been created and is available.
+/// ```
+/// # fn main() -> anyhow::Result<()> {
+/// # use momento_test_util::create_doctest_cache_client;
+/// # tokio_test::block_on(async {
+/// # let (cache_client, cache_name) = create_doctest_cache_client();
+/// use momento::requests::cache::scalar::keys_exist::KeysExist;
+/// use momento::requests::cache::scalar::keys_exist::KeysExistRequest;
+///
+/// let request = KeysExistRequest::new(
+///     cache_name,
+///     vec!["key1", "key2", "key3"]
+/// );
+///
+/// let result = cache_client.send_request(request).await?;
+/// if !result.exists.is_empty() {
+///     println!("Processing list of booleans: {:#?}", result.exists);
+/// } else {
+///     println!("Received empty list of booleans!");
+/// }
+/// # Ok(())
+/// # })
+/// # }
+/// ```
+pub struct KeysExistRequest<K: IntoBytes> {
+    cache_name: String,
+    keys: Vec<K>,
+}
+
+impl<K: IntoBytes> KeysExistRequest<K> {
+    pub fn new(cache_name: impl Into<String>, keys: Vec<K>) -> Self {
+        Self {
+            cache_name: cache_name.into(),
+            keys,
+        }
+    }
+}
+
+impl<K: IntoBytes> MomentoRequest for KeysExistRequest<K> {
+    type Response = KeysExist;
+
+    async fn send(self, cache_client: &CacheClient) -> MomentoResult<KeysExist> {
+        // consume self.keys once to convert all keys to bytes
+        let byte_keys: Vec<Vec<u8>> = self.keys.into_iter().map(|key| key.into_bytes()).collect();
+
+        // convert keys to strings for the response exists_dictionary because HashMap<IntoBytes, bool> is not allowed
+        let string_keys: Vec<String> = byte_keys
+            .iter()
+            .map(|key| parse_string(key.clone()))
+            .collect::<MomentoResult<Vec<String>>>()?;
+
+        let request = prep_request_with_timeout(
+            &self.cache_name,
+            cache_client.configuration.deadline_millis(),
+            momento_protos::cache_client::KeysExistRequest {
+                cache_keys: byte_keys,
+            },
+        )?;
+
+        let response = cache_client
+            .data_client
+            .clone()
+            .keys_exist(request)
+            .await?
+            .into_inner();
+
+        Ok(KeysExist {
+            exists: response.exists.clone(),
+            exists_dictionary: string_keys
+                .into_iter()
+                .zip(response.exists.clone())
+                .collect(),
+        })
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct KeysExist {
+    pub exists: Vec<bool>,
+    pub exists_dictionary: HashMap<String, bool>,
```

Comment:
was unable to get HashMap<IntoBytes, bool> working, but got a String version working. Can probably add Vec<u8> version as well if that seems useful

Reply:
I think string version is great for now, thx

Reply:
rather than exposing this as another property on the struct, could we expose it via an `into`/`From` thing? and show it in the docstring example? I think you will need to save the keys into the struct in order to support that, but it could be a private property rather than a public one?

Just thinking about trying to make the patterns for accessing these different shapes of the data consistent from one API to the next, and e.g. for Get, we mostly expose alternate types via `into`. WDYT?

Reply:
that makes sense, made those fields private and implemented `From` in https://github.com/momentohq/client-sdk-rust/pull/217/commits/11c6c6d7092def3089a3a5b0c8bd12d7a31b3b68 

should I keep the `exists` and `exists_dictionary` method? or remove those?

Reply:
I think for now let's remove them and just show examples of the `into`s. Always easier to add more API surface area later than to remove it! :)